### ISSUE TITLE
Public Server Availability and Reliability

### ISSUE THREAD

**Issue by devfans** [opentimestamps/opentimestamps-server#71]
Hi,

I want to know if the public server is reliable for production usage. Like if we have 100,000 entries generated every day and use the public server to generate proofs for each of them, is it proper? From my understanding, the public server is already aggregating data, so wondering do we still need to aggregating our daily data in batch and timestamp it once every day.
Thanks

**Comment by petertodd** [comment]
On Sun, Dec 01, 2019 at 10:18:21PM -0800, Stefan Liu wrote:
> Hi,
> 
> I want to know if the public server is reliable for production usage. Like if we have 100,000 entries generated every day and use the public server to generate proofs for each of them, is it proper? From my understanding, the public server is already aggregating data, so wondering do we still need to aggregating our daily data in batch and timestamp it once every day.
> Thanks

100,000 entries per day is absolutely fine. As you said, the servers do
aggregation so the per-timestamp-cost is just bandwidth and CPU time.

Now, if you intend to do significantly more than that - say 1,000,000 entries
per day - it'd be appreciated if you aggregated your own timestamp requests to
reduce our bandwidth/CPU usage. The easiest way to do that is with @rcasatta's
foxglove aggregator: https://github.com/opentimestamps/foxglove

Basically, you run it locally, and it builds "mini" merkle trees two or three
times per second, with the tip of those merkle trees being submitted to the
public servers.

**Comment by petertodd** [comment]
BTW we're now using a third-party service to monitoring OTS calendar uptime: https://uptime.opentimestamps.net/

This isn't perfect, as it isn't smart enough to check if the timestamps returned are actually correct. But 99% of the time if it returns a valid OK 200 response it is, so that site is pretty close.
